src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -1434,85 +1080,14 @@ private LambdaTranslationContext analyzeLambda(JCLambda tree, String statKey) {
 
         @Override
         public void visitMethodDef(JCMethodDecl tree) {
-            List<ClassSymbol> prevTypesUnderConstruction = typesUnderConstruction;
             List<Frame> prevStack = frameStack;
             try {
-                if (TreeInfo.isConstructor(tree))       // start early construction context (Object() notwithstanding)
-                    typesUnderConstruction = typesUnderConstruction.prepend(currentClass());
                 frameStack = frameStack.prepend(new Frame(tree));
                 super.visitMethodDef(tree);
-            } finally {
-                frameStack = prevStack;
-                typesUnderConstruction = prevTypesUnderConstruction;
-            }
-        }
-
-        @Override
-        public void visitNewClass(JCNewClass tree) {
```

Comment:
most of the removed visitor methods had to do with the `typesUnderConstruction` workaround, which has now been removed